aten/src/ATen/cuda/CusolverSpHandlePool.cpp

```diff
@@ -0,0 +1,51 @@
+#include <ATen/cuda/CUDAContext.h>
+#include <ATen/cuda/detail/DeviceThreadHandles.h>
+
+#ifdef CUDART_VERSION
+
+namespace at { namespace cuda {
+namespace {
+
+void createCusolverSpHandle(cusolverSpHandle_t *handle) {
+  TORCH_CUSOLVER_CHECK(cusolverSpCreate(handle));
+}
+
+void destroyCusolverSpHandle(cusolverSpHandle_t handle) {
```

Comment:
Note to reviewers:
This file was created as a copy of similar files in the folder implementing pools of handles.